src/main/java/org/terasology/alterationEffects/regenerate/RegenerationAlterationEffect.java

```diff
@@ -124,12 +126,13 @@ else if (!modifiersFound || !effectModifyEvent.getHasInfDuration()) {
      *
      * @param instigator    The entity who applied the regen effect.
      * @param entity        The entity that the regen effect is being applied on.
-     * @param id            Inapplicable to the regen effect.
+     * @param id            Id is used for unique identification in regen scheduler.
      * @param magnitude     The magnitude of the regen effect.
      * @param duration      The duration of the regen effect.
      */
     @Override
     public void applyEffect(EntityRef instigator, EntityRef entity, String id, float magnitude, long duration) {
-        applyEffect(instigator, entity, magnitude, duration);
+        //applyEffect(instigator, entity, magnitude, duration);
+        entity.send(new ActivateRegenEvent(id, (int) magnitude, ((float) duration) / 1000));
```

Comment:
Slight nitpick here. Do we want to round down, round up or use natural rounding?

Ie, if we use this method twice with 0.3 & 0.7 what should the event send?  
Down →0 & 0 (this is the current method)
Natural →0 & 1
Up →1 & 1

I can see legitimate arguments for all of these. I would personally go for natural but it's up to you which one you pick. Just make sure you've considered the options.


Additionally, we can avoid sending an ActivateRegenEvent when the magnitude is 0 because it just won't do anything correct?

Reply:
I just realized that the `RegenComponent` is saving the regen value in floats. But the event does not take floats. Calls for an update in the event, which would resolve this issue, leaving it for regen scheduler to handle